File: Binarygap/binaryGap.js
function binaryGap(number){
   const binaryValues = number.toString(2)
   let currentGap = 0
   let maximumGap = 0

    for(i=0; i<binaryValues.length; i++){
        if(binaryValues[i] === "0"){
            currentGap++
        }else{
            if(currentGap>maximumGap){
                maximumGap = currentGap
            }
            currentGap = 0
        }
    }
    console.log(`the maximum gap is ${maximumGap}`)
}
binaryGap(529)

File: Binarygap/binary_gap.py
def binary_gap(number):
    binary_values = bin(number)[2:]
    current_gap = 0
    maximum_gap = 0
    for binary_value in binary_values:
        if binary_value == "0":
            current_gap += 1
        else:
            if current_gap > maximum_gap:
                maximum_gap = current_gap
            current_gap = 0
    print(f"maximum gap is {maximum_gap}")
    return maximum_gap
binary_gap(529)